docs/DafnyRef/Grammar.md

```diff
@@ -183,17 +183,12 @@ by IDEs and documentation generation tools to present information to users.
 
 In Dafny programs.
 * Documentation comments (a) either begin with `/**` or (b) begin with `//` or /*` in specific locations
-* Doc-comments may be associated with any declaration, including type definitions, export declarations, and datatype constructors.
-* They may be placed before or after the declaration. 
-   * If before, it must be a `/**` comment and may not have any blank or white-space lines between the comment
-     and the declaration.
-   * If after, any comments are placed after the signature (with no intervening lines), but before any
-     specifications or left-brace that starts a body, and may be `//` or `/**` or `/*` comments.
-   * If doc-comments are in both places, only the comments after the declaration are used.
-* Doc-comments after the declaration are preferred.
-* If the first of a series of single-line or multi-line comments is interpreted as a doc-string, then any subsequent comments
-  are appended to it, so long as there are no intervening lines, whether blank, all white-space or containing program text.
-* The extraction of the doc-string from a multiline comment follow these rules 
+* Documentation comments may be associated with any declaration, including type definitions, export declarations, and datatype constructors.
+* They may be placed before the declaration, between the declaration and the definition or after the definition. The priority of these comments is the following:
+   * If there is a comment starting with `/**` right before the definition, it's the documentation comment.
+   * Otherwise, if one or more comments are between a declaration and the definition (which is not usual in other programming languages), whether it begins with `//` or `/**` or `/*`, they are considered as the documentation comment, as there is no other reason a comment would be at this place.
```

Comment:
```suggestion
   * Otherwise, if one or more comments are between a declaration and the definition (which is not usual in other programming languages), if it begins with `//` or `/**` or `/*`, they are considered as the documentation comment, as there is no other reason a comment would be at this place.
```

Reply:
Actually, I would prefer to keep "whether" and use plural because it refers to one or more comments, and all comments start at least with one of the given tokens.